src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java

```diff
@@ -0,0 +1,70 @@
+package me.totalfreedom.totalfreedommod.command;
+
+import me.totalfreedom.totalfreedommod.admin.Admin;
+import me.totalfreedom.totalfreedommod.player.FPlayer;
+import me.totalfreedom.totalfreedommod.rank.Displayable;
+import me.totalfreedom.totalfreedommod.rank.Rank;
+import me.totalfreedom.totalfreedommod.util.FLog;
+import me.totalfreedom.totalfreedommod.util.FUtil;
+import org.apache.commons.lang.StringUtils;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+@CommandPermissions(level = Rank.SUPER_ADMIN, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Vanish yourself.", usage = "/<command>")
+public class Command_vanish extends FreedomCommand
+{
+
+    @Override
+    public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+        final Displayable display = plugin.rm.getDisplay(playerSender);
+        String loginMsg = display.getColoredLoginMessage();
+        String displayName = display.getColor() + playerSender.getName();
+        Admin admin = getAdmin(playerSender);
+        final FPlayer fPlayer = plugin.pl.getPlayer(playerSender);
+        if (fPlayer.isVanish())
+        {
+            msg(ChatColor.GOLD + "You have been unvanished.");
+            if (admin.hasLoginMessage())
+            {
+                loginMsg = FUtil.colorize(admin.getLoginMessage());
+            }
+            FUtil.bcastMsg(ChatColor.AQUA + playerSender.getName() + " is " + loginMsg);
```

Comment:
Im not sure on this. Is this not different depending on the rank / title the admin has?